src/ray/core_worker/common.h

```diff
@@ -13,11 +13,15 @@ namespace ray {
 using WorkerType = rpc::WorkerType;
 
 /// Information about a remote function.
-struct RayFunction {
+class RayFunction {
+ public:
+  RayFunction() {}
+  RayFunction(Language language, std::vector<std::string> function_descriptor)
```

Comment:
the function descriptor should be passed as `const std::vector<std::string>&`, otherwise an additional copy might be made.